pipeline/download_gene.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2021, Blue Brain Project, EPFL
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script that download and postprocess ISH dataset from Allen Brain."""
+from __future__ import annotations
+
+import argparse
+import logging
+import sys
+from pathlib import Path
+
+import numpy as np
+import PIL
+
+
+logger = logging.getLogger("download-gene")
+
+
+def parse_args():
+    """Parse command line arguments.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        The parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--gene-name",
+        type=str,
+        help="""\
+        Path to CCFv3 annotation volume.
+        """,
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        help="""\
+        Path to output directory to save results.
+        """,
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def postprocess_dataset(dataset):
+    """Post process given dataset.
+
+    Parameters
+    ----------
+    dataset : iterable
+        List containing dataset, each element being tuple
+        (img_id, section_coordinate, img, df)
+
+    Returns
+    -------
+    dataset_np : np.ndarray
+        Array containing gene expressions of the dataset.
+    expression_np : np.ndarray
+        Array containing image expression of the dataset.
+    metadata_dict : dict
+        Dictionary containing metadata of the dataset.
+        Keys are section numbers and image ids.
+    """
+    metadata_dict = {}
+    section_numbers = []
+    image_ids = []
+    expression_np = []
+    dataset_np = []
+
+    for img_id, section_coordinate, img, img_expression, df in dataset:
```

Comment:
Either logging or a progress bar would be nice here since it will take a lot of time.

If you wanna go for a progress bar you will need to provide `total` to tqdm since it is we iterate over a generator (the length cannot be implied). 

However, the `n_images` will be know after fetching the metadata and can be accessed via `len(dataset_downloader)`.

We did it here https://github.com/BlueBrain/Atlas-Download-Tools/blob/b08093f5bebf53c7491ce1b314382eeb0e78d656/src/atldld/cli/download.py#L96

Reply:
Done in [9bc2e8d](https://github.com/BlueBrain/Deep-Atlas/pull/8/commits/9bc2e8d8ca1a6a9c1ef80f94c923e2d142ec3a84).

Reply:
Amazing. Thank you!